### DISCUSSION TITLE
Error XAPRAS7023 : System.IO.DirectoryNotFoundException: Could not find a part of the path

### DISCUSSION THREAD

**Discussion by dawnlyn** [AvaloniaUI/Avalonia#17250] [Q&A]
I created a new project using the Avalonia Cross Platform Application template, but I found that as soon as I put the xaml file in the Android project at the same level as MainActivity.cs, I get an error:
```
Error XAPRAS7023 : System.IO.DirectoryNotFoundException: Could not find a part of the path 
'D:\Work\JiuZhouUniverse\Test\Test\Test.Android\obj\Debug\net9.0-android\Avalonia\Avalonia\Test.Android.dll'.
```
He will have an extra level of directories, but other projects are not affected.

Not having developed in Xamarin, I'm not sure how this should be solved and where I can find the corresponding documentation?

Thank you for your help.

This problem occurs in 11.1.4, .Net9 preview version

Translated with www.DeepL.com/Translator (free version)

**Comment by dawnlyn** [comment]
After I tried upgrading to 11.2.0-rc1, it seems that the problem is solved and he doesn't report errors anymore!

So this is actually a bug, right?

**Comment by JeromeTalan** [comment]
I have the the same error, even if i upgrade to 11.2. Copy the files to the subdir works

XAPRAS7023
'...\bin\Debug\net8.0\android-x86\Avalonia ODS VideoManager.dll'.
   à System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   à System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   à System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   à Xamarin.Android.Tasks.MonoAndroidHelper.HasMonoAndroidReference(ITaskItem assembly)
   à Xamarin.Android.Tasks.ProcessAssemblies.SetMetadataForAssemblies(List`1 output, Dictionary`2 symbols)
   à Xamarin.Android.Tasks.ProcessAssemblies.RunTask()
   à Microsoft.Android.Build.Tasks.AndroidTask.Execute() dans /Users/runner/work/1/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/AndroidTask.cs:ligne 25	Avalonia ODS VideoManager.Android	C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\34.0.113\targets\Microsoft.Android.Sdk.AssemblyResolution.targets	123

**Comment by BobbyCannon** [comment]
This is still an issue. Everything was working fine with 11.2.0, but when upgrading to 11.2.1 the XAPRAS7023 issue came back.

`\Debug\net9.0-android\**Avalonia\Avalonia**\Test.Android.dll`

I think the problem is the double "Avalonia\Avalonia" in the path...

**Comment by rubiomiguel06** [comment]
I've just faced this issue on 11.2.5.. 

As @JeromeTalan mentioned, copying manually the files to the faulty path makes it work, but this it not a solution.
To me, this happens only when I add an `<AvaloniaResource>` inside the .Android project. If I remove all AvaloniaResources it works well
